src/tabcenter.js

```diff
@@ -99,9 +99,22 @@ TabCenter.prototype = {
     for (let identity of identities) {
       const identityItem = document.createElement("div");
       identityItem.className = "newtab-menu-identity";
-      identityItem.addEventListener("mouseup", () => {
-        this.hideNewTabMenu();
-        browser.tabs.create({ cookieStoreId: identity.cookieStoreId });
+      identityItem.addEventListener("mouseup", async e => {
+        switch (e.which) {
+        case 1:
```

Comment:
I think merging and then refactoring can reduce confusion for review, there will be multiple pull requests call browser.tabs.create to next tab.